packages/webdriverio/src/middlewares.js

```diff
@@ -15,7 +15,19 @@ export const elementErrorHandler = (fn) => (commandName, commandFn) => {
             this.elementId = element.elementId
 
             try {
-                return await fn(commandName, commandFn).apply(this, args)
+                const result = await fn(commandName, commandFn).apply(this, args)
+
+                /**
+                 * assume Safari responses like { error: 'no such element', message: '', stacktrace: '' }
+                 * as `stale element reference`
+                 */
+                if (result && result.error === 'no such element') {
```

Comment:
fix #3645

Reply:
Can we handle this inside webdriver instead?
packages/webdriver/src/utils.js
isSuccessfulResponse (statusCode, body) {

Reply:
ultimately, it would be nice if we could consume this inCustomRequestError and treat safari's stale elements the same as the rest

Reply:
@abjerstedt sorry to say there is no way to do it.
We have to keep this dirty hack.
The problem is that there is no difference between Safari's not found element and Safari's stale element exception.

My proposal would be to leave this code as is.

Reply:
@mgrybyk We still need to handle the safari stales upstream in webdriver (but i can put a diff pr up). 

We need to block the triple requests and just return an element not found. This should hypothetically cause the middleware to just work.

Reply:
Agree, I've noticed that requests are multiplied.
If you can assist it would be really awesome!